corehq/form_processor/backends/couch/casedb.py

```diff
@@ -68,3 +69,19 @@ def post_process_case(self, case, xform):
 
     def get_reverse_indexed_cases(self, case_ids):
         return get_reverse_indexed_cases(self.domain, case_ids)
+
+    def filter_closed_extensions(self, extensions_to_close):
+        # filter out cases that are closed which we already have cached
+        extensions_to_close = [
+            case_id for case_id in extensions_to_close
+            if case_id not in self.cache or not self.cache[case_id].closed
+        ]
+        if extensions_to_close:
+            # filter out any other cases that are already closed (or deleted)
+            closed_deleted = [
+                case_id for case_id, _, _ in
+                CaseAccessorCouch.get_closed_and_deleted_ids(self.domain, extensions_to_close)
+            ]
+            extensions_to_close = [case_id for case_id in extensions_to_close if case_id not in closed_deleted]
+
+        return extensions_to_close
```

Comment:
👍 
I think it would be totally fair to not support couch domains for extension cases any more, after `livequery`.